### PR TITLE
[CARBONDATA-2757][BloomDataMap]  Fix bug when building bloomfilter on measure column

### PR DESCRIPTION
**Pull request by kevinjmh**
1. support to get raw data from decimal column page when building datamap in loading process

2. convert decimal column to java datatype when rebuilding bloom datamap from query result

3. convert boolean to byte as carbon wants

4. fix bugs when measure column is null

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7306/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6073/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7312/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6077/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5920/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5924/

File: core/src/main/java/org/apache/carbondata/core/datastore/page/DecimalColumnPage.java [hunk 1]
```diff
@@ -106,4 +109,48 @@ public void setDoublePage(double[] doubleData) {
     throw new UnsupportedOperationException("invalid data type: " + dataType);
   }
 
+  private BigDecimal getDecimalFromRawData(int rowId) {
```
Comment (xuchuanyin) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/DecimalColumnPage.java]: Please add comment to describe when to use this method and the below method

File: datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/AbstractBloomDataMapWriter.java [hunk 2]
```diff
@@ -129,8 +130,12 @@ protected void addValue2BloomIndex(int indexColIdx, Object value) {
     // convert non-dict dimensions to simple bytes without length
     // convert internal-dict dimensions to simple bytes without any encode
     if (indexColumns.get(indexColIdx).isMeasure()) {
-      if (value == null) {
-        value = DataConvertUtil.getNullValueForMeasure(indexColumns.get(indexColIdx).getDataType());
+      // NULL value of all measures are already processed in `ColumnPage.getData`
+      // or `RawBytesReadSupport.readRow` with actual data type
+
+      // Carbon stores boolean as byte. Here we convert it for `getValueAsBytes`
+      if (value instanceof Boolean) {
```
Comment (xuchuanyin) [inline datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/AbstractBloomDataMapWriter.java]: can we use the datatype as the condition?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7351/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6115/

File: datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/AbstractBloomDataMapWriter.java (hunk 2 above)
Comment (kevinjmh) [reply datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/AbstractBloomDataMapWriter.java]: Yes. FYI, column of boolean type uses column page with inner datatype Byte, and some disagreement on DataType exists between `getData` and `getNull`.   I will changed that.

File: core/src/main/java/org/apache/carbondata/core/datastore/page/DecimalColumnPage.java (hunk 1 above)
Comment (kevinjmh) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/DecimalColumnPage.java]: Fixed

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7355/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6119/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5936/

**Comment by xuchuanyin** [comment]
LGTM

**Pull request closed without merging**